<!-- IMU message attitude to TF relay (with the option to publish odom messages as well
     Warning: Our IMU's typpicaly cannot estimate translation (except for the fancy MTi-G 
     and similar with GPS. This relay fills the translation fields with zeros.
-->
<launch>
  <node pkg="norlab_imu_tools" type="imu_and_fix_to_tf_node" name="imu_and_fix_to_tf_node" output="screen">
    <remap from="imu_topic" to="/mti/sensor/imu" />
    <remap from="gps_odom_topic" to="/odom_utm" />
    <param name="odom_frame" value="world" />
    <param name="base_frame" value="imu_link" />
    <param name="publish_gps_translation" value="False" />
    <param name="apply_gps_heading_correction" value="True" />
    <param name="gps_heading_correction_weight" value="0.1" />
    <param name="gps_heading_min_dist" value="1.5" />

    ## This rosparam allows you to set IMU alignment w.r.t. base_link
    ## E.g. in our peli-case, the IMU is mounted upside-down, so set the RPY appropriately
    ## Values in RAD 
    <rosparam param="imu_alignment_rpy">[3.14159265359, 0.0, 3.14159265359]</rosparam>  <!--Pelicase box for skidoo, all angles in radians! -->
    <!-- from test in engineering  93.0953deg (1.62482rad),), actually I used 1.570796327.-->
    <!-- from test in pouliot  106.192deg (1.8534rad) -->
    <param name="mag_north_correction_yaw" value="1.8534" /> <!-- 15.66 deg == 15Â°40'  => 0.27331833              1.840796327 -> 15.66 deg + 90 deg -->


    ## If odom message needed (but keep in mind that only the orientation makes sense):
    <param name="publish_odom" value="True" />
    <param name="odom_topic_name" value="imu_odom" />

  </node>

  <node pkg="gps_common" type="utm_odometry_node" name="gps_to_umt_odom_node" output="screen">
    <remap from="odom" to="odom_utm" />
    <param name="frame_id" value="gps_world" />
    <param name="child_frame_id" value="gps_base_link" />
  </node>

</launch>
